tests/unit/client.py

```diff
@@ -0,0 +1,659 @@
+        x = ClientBase(ep, 'some_base64_string')
+        y = x.object_url('abc', '123', 'xy23z')
+        assert y == 'http://127.0.0.1:8000/abc/123/xy23z'
+
+# For testing the client library we need a running HIL server, with dummy
+# objects populated. Following classes accomplish that end.
+# It shall:
+#       1. Configures haas.cfg
+#       2. Instantiates a database
+#       3. Starts a server on an arbitary port
+#       4. Populates haas with dummy objects
+#       5. tears down the setup in a clean fashion.
+
+
+# pytest.fixture(scope="module")
+
+def make_config():
+    """ This function creates haas.cfg with desired options
+    and writes to a temporary directory.
+    It returns a tuple where (tmpdir, cwd) = ('location of haas.cfg', 'pwdd')
+    """
+    tmpdir = tempfile.mkdtemp()
+    cwd = os.getcwd()
+    os.chdir(tmpdir)
+    with open('haas.cfg', 'w') as f:
+        config = '\n'.join([
+            '[general]',
+            '[devel]',
+            'dry_run=True',
+            '[auth]',
+            'require_authentication = True',
+
+            '[headnode]',
+            'base_imgs = base-headnode, img1, img2, img3, img4',
+            '[database]',
+            'uri = sqlite:///%s/haas.db' % tmpdir,
+            '[extensions]',
+            'haas.ext.auth.database =',
+            'haas.ext.switches.mock =',
+            'haas.ext.switches.nexus =',
+            'haas.ext.switches.dell =',
+            'haas.ext.switches.brocade =',
+            'haas.ext.obm.mock =',
+            'haas.ext.obm.ipmi =',
+            'haas.ext.network_allocators.vlan_pool =',
+            '[haas.ext.network_allocators.vlan_pool]',
+            'vlans = 1001-1040',
+
+        ])
+        f.write(config)
+        return (tmpdir, cwd)
+
+
+def cleanup((tmpdir, cwd)):
+    """ Cleanup crew, when all tests are done.
+    It will shutdown the haas server,
+    delete any files and folders created for the tests.
+    """
+
+    os.remove('haas.cfg')
+    os.remove('haas.db')
+    os.chdir(cwd)
+    os.rmdir(tmpdir)
+
+
+def initialize_db():
+    """ Creates an  database as defined in haas.cfg."""
+    check_call(['haas-admin', 'db', 'create'])
+    check_call(['haas', 'create_admin_user', username, password])
+
+
+def run_server(cmd):
+    """This function starts a haas server.
+    The arguments in 'cmd' will be a list of arguments like required to start a
+    haas server like ['haas', 'serve', '8000']
+    It will return a handle which can be used to terminate the server when
+    tests finish.
+    """
+    proc = Popen(cmd)
+    return proc
+
+
+def populate_server():
+    """
+    Once the server is started, this function will populate some mock objects
+    to faciliate testing of the client library
+    """
+    sess = requests.Session()
+    sess.auth = (username, password)
+
+    # Adding nodes, node-01 - node-06
+    url_node = 'http://127.0.0.1:8000/node/'
+    api_nodename = 'http://schema.massopencloud.org/haas/v0/obm/'
```

Comment:
I defined the variable `api_nodename` because, using it everytime would exceed the character width of 79 and violate pep8. To reduce unnecessary lines of code I defined it as a variable and used it. Later I made it a for loop and hence it was used only once. I have renamed the variable name from `api_nodename` to `ipmi`. This is better than using a string literal for future tests if we have a case of registering nodes with other obm types. Next commit has the changes.